File: samples/csharp/PdfToText.cs
using System;
using SelectPdf.Api;

namespace Samples
{
    public class PdfToText
    {
        public static void RunTest()
        {
            string testUrl = "https://selectpdf.com/demo/files/selectpdf.pdf";
            string testPdf = "Input.pdf";
            string localFile = "Result.txt";
            string apiKey = "Your API key here";

            Console.WriteLine("This is SelectPdf-{0}.", ApiClient.CLIENT_VERSION);

            try
            {
                PdfToTextClient client = new PdfToTextClient(apiKey);

                // set parameters - see full list at https://selectpdf.com/pdf-to-text-api/
                client
                    .setStartPage(1) // start page (processing starts from here)
                    .setEndPage(0) // end page (set 0 to process file til the end)
                    .setOutputFormat(OutputFormat.Text) // set output format (0-Text or 1-HTML)
                ;

                Console.WriteLine("Starting pdf to text ...");

                // convert local pdf to local text file
                client.getTextFromFileToFile(testPdf, localFile);

                // extract text from local pdf to memory
                // string text = client.getTextFromFile(testPdf);
                // print text
                // Console.WriteLine(text);

                // convert pdf from public url to local text file
                // client.getTextFromUrlToFile(testUrl, localFile);

                // extract text from pdf from public url to memory
                // string text = client.getTextFromUrl(testUrl);
                // print text
                // Console.WriteLine(text);

                Console.WriteLine("Finished! Number of pages processed: {0}.", client.getNumberOfPages());

                // get API usage
                UsageClient usageClient = new UsageClient(apiKey);
                UsageInformation usage = usageClient.getUsage(false);
                Console.WriteLine("Conversions remained this month: {0}.", usage.Available);
            }
            catch (Exception ex)
            {
                Console.WriteLine("An error occurred: " + ex.Message);
            }
        }

    }
}

File: samples/vbnet/PdfToText.vb
Imports System
Imports SelectPdf.Api

Public Class PdfToText
    Public Shared Sub RunTest()
        Dim testUrl As String = "https://selectpdf.com/demo/files/selectpdf.pdf"
        Dim testPdf As String = "Input.pdf"
        Dim localFile As String = "Result.txt"
        Dim apiKey As String = "Your API key here"

        Console.WriteLine("This is SelectPdf-{0}.", ApiClient.CLIENT_VERSION)

        Try
            Dim client As PdfToTextClient = New PdfToTextClient(apiKey)

            ' set parameters - see full list at https://selectpdf.com/pdf-to-text-api/

            client.setStartPage(1) ' start page (processing starts from here)
            client.setEndPage(0) ' End page (Set 0 To process file til the End)
            client.setOutputFormat(OutputFormat.Text) ' Set output format (0-Text Or 1-HTML)

            Console.WriteLine("Starting pdf to text ...")

            ' convert local pdf to local text file
            client.getTextFromFileToFile(testPdf, localFile)

            ' extract text from local pdf to memory
            ' Dim text As String = client.getTextFromFile(testPdf)
            ' print text
            ' Console.WriteLine(text)

            ' convert pdf from public url to local text file
            ' client.getTextFromUrlToFile(testUrl, localFile)

            ' extract text from pdf from public url to memory
            ' Dim text As String = client.getTextFromUrl(testUrl)
            ' print text
            ' Console.WriteLine(text)

            Console.WriteLine("Finished! Number of pages processed: {0}.", client.getNumberOfPages())

            ' get API usage
            Dim usageClient As UsageClient = New UsageClient(apiKey)
            Dim usage As UsageInformation = usageClient.getUsage(False)
            Console.WriteLine("Conversions remained this month: {0}.", usage.Available)

        Catch ex As Exception
            Console.WriteLine("An error occurred: " & ex.Message)
        End Try
    End Sub
End Class
